### PR TITLE
feat: update PATCH /connectors/:id

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

update PATCH /connectors/:id (now this API can be use to update `sync_profile`)
also, update users' profile when signing in as long as `sync_profile` is true 

## Testing

UTs and ITs.

### CODE REVIEW

File: packages/core/src/routes/session/social.ts [hunk 1]
```diff
@@ -100,10 +100,12 @@ export default function socialRoutes<T extends AnonymousRouter>(router: T, provi
       ctx.log(type, { userId: id });
 
       const { name, avatar } = userInfo;
-      const profileUpdate = {
-        name: syncProfile && name ? name : undefined,
-        avatar: syncProfile && avatar ? avatar : undefined,
-      };
+      const profileUpdate = Object.fromEntries(
+        Object.entries({
+          name: syncProfile && name ? name : undefined,
```
Comment (wangsijie) [inline packages/core/src/routes/session/social.ts:105]: use `conditional`

File: packages/core/src/routes/connector.ts [hunk 2]
```diff
@@ -128,6 +128,34 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+        params: { id },
+        body,
+      } = ctx.guard;
+
+      const { type, metadata } = await getLogtoConnectorById(id);
+
+      if (body.syncProfile) {
+        assertThat(
+          type === ConnectorType.Social,
+          new RequestError({ code: 'connector.unexpected_type', status: 422 })
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:148]: This error type can be more specific?

**Review by wangsijie — COMMENTED** [review]
can we move this to `'/connectors/:id'? as it seems not special.

File: packages/core/src/routes/session/social.test.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
+/* eslint-disable max-lines */
```
Comment (wangsijie) [inline packages/core/src/routes/session/social.test.ts:1]: can we split this file?

Comment (darcyYe) [reply packages/core/src/routes/session/social.test.ts]: Removed duplicated UT of `POST /session/register/social` (it has already been moved to packages/core/src/routes/session/social.bind-social.test.ts but forgot to be removed).

**Review by wangsijie — APPROVED** [review]

**Pull request merged**